### PR TITLE
feat: calc custom start time in case schedule already ended

### PR DESCRIPTION
**Pull request by jorijnsmit**
we happened to encounter this use case this week; it is possible that when setting the schedule the previous one ended already. start time then has to be adjusted to the next cycle or later, since emissions will not be processed retroactively by the tree.

didnt write tests for this yet.

### CODE REVIEW

File: contracts/VotiumBribesProcessor.sol [hunk 1]
```diff
@@ -129,10 +130,22 @@ contract VotiumBribesProcessor is CowSwapSeller {
...
-    function _lastSchedule(address _token) internal view returns (uint256 endingTime) {
+    function _getScheduleTimes(address _token) internal view returns (
+        uint256 newStartTime, uint256 newEndTime, uint256 duration
+    ) {
         IRewardsLogger.UnlockSchedule[] memory schedules = REWARDS_LOGGER
             .getUnlockSchedulesFor(address(BVE_CVX), _token);
-        endingTime = schedules[schedules.length - 1].end;
+        uint256 lastEndingTime = schedules[schedules.length - 1].end;
+        uint256 minStartTime = BADGER_TREE.lastPublishTimestamp();
+        if(minStartTime <= lastEndingTime) {
```
Comment (petrovska-petro) [inline contracts/VotiumBribesProcessor.sol:140]: i envisioned will be the condition on the opposite way, say:

- if we run `swapCVXTobveCVXAndEmit` & `emitBadger` way before the **lastEndingTime** (which may be the case if bribes are process promptly), then the starting time should be **lastEndingTime**, otherwise running before the last schedule ended two schedules will overlap. Since:

```
newStartTime = minStartTime + 4 hours;
```

may be way before than `lastEndingTime`.

Then, if when running these two methods and `minStartTime` is **greater** than `lastEndingTime` means that schedules were delayed for some reason and we should post the schedules with the `starting time = minStartTime + GRACE_CYCLE_TIME` (being the grace cycle time 4h).

Another case, perhaps to check is when `minStartTime + 2h` may fall within the `lastEndingTime`, since then some portions of the rewards there will not be distributed properly if we use `lastEndingTime` as `newStartTime`.

Perhaps, another checkup worth adding is for example in the `minStartTime <= lastEndingTime`,

**Comment by jorijnsmit** [comment]
added some simple asserts.

there is a draft for a proper test of this feature but im afraid it cannot be run; fastforwarding the badger tree is a very elaborate process.

**Comment by GalloDaSballo** [comment]
@jorijnsmit @petrovska-petro I think in theory code is fine, we may also just want to receive the "desired timestamp" and then run validation to ensure it's within 14 days.

This will avoid potential issues if the math ends up having edge cases

As for the test, you can use: https://etherscan.io/address/0xbe82a3259ce427b8bcb54b938b486dc2af509cc3#code#L1571

After proposing root (perhaps the same data) so that you update the timer
